sdk/core/azure-core/src/main/java/com/azure/android/core/http/responsepaging/PagedDataResponseCollection.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package com.azure.android.core.http.responsepaging;
+
+import com.azure.android.core.http.Response;
+import com.azure.android.core.util.paging.Page;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Represents a collection of pages where page and the response for retrieving it can be returned synchronously
+ * @param <T> type of the items contained in the page
+ * @param <P> page that contains the items
+ */
+public class PagedDataResponseCollection<T, P extends Page<T>> {
```

Comment:
Is this the sync version of `AsyncPagedDataCollection`? If so, they should have similar names. E.g. `PagedDataCollection`.

Reply:
There is a PagedDataCollection in util.paging. We can rename AsyncPagedDataCollection to AsyncPagedDataResponseCollection

Reply:
Okay, I understand why they are the way they are, no need to rename.